### ISSUE TITLE
Convert 'px' values to 'rem' to ensure the guidance site appears correctly when using different browser font sizes

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-design-system#217]
## Summary
(This ticket is essentially the same as mi6/ic-ui-kit#290 but it is about applying the changes to the code in the guidance site.)

A lot of spacing and sizing values in the CSS across the guidance site use the `px` unit. This means that some of the site is rendered incorrectly when the font size in the user's browser is changed to something other than the default. Using `rem` instead would make elements within the site resize and render correctly at different font sizes.

## 💬 Description

When the font size in the browser is changed, some of the guidance site does not resize correctly.

Examples of some of the most obvious issues:
- The 'Show navigation section' button (which appears on smaller screens) stays the same size when the browser font size is changed:

<img src="https://user-images.githubusercontent.com/111882109/212397967-d774e15c-3f62-4f45-a8da-d2fbe05f90c5.png" width="400">

- Icon picker - text size on the left and icon sizes:

<img src="https://user-images.githubusercontent.com/111882109/212397406-a3ad3a37-3758-4da7-b179-92f57f1693d2.png" width="400">

After some exploration, it appears that the cause is using the absolute `px` unit for a lot of the spacing and sizing in the CSS across the site - `rem` is a relative unit so using this for elements will make them scale to match the chosen font size.

**Note 1:** I believe it should be almost all `px` values that need changing across all of the code for the guidance site. Anything to do with breakpoints (e.g. media queries) would need to remain, but I would expect that every other `px` value would need converting. I think this should probably include values in the component examples as well e.g. [SVG heights and widths](https://github.com/mi6/ic-design-system/blob/37dbe3e7d94bc2d14e3686d745149a3a81fa5efa/src/content/structured/components/top-nav/code.mdx?plain=1#L87), [inline styles](https://github.com/mi6/ic-design-system/blob/37dbe3e7d94bc2d14e3686d745149a3a81fa5efa/src/content/structured/components/skeleton/code.mdx?plain=1#L83).

**Note 2:** Converting `px` to `rem` _should_ fix all issues that occur when using a different font size, but it will be important to look through the guidance site after doing this to make sure that there aren't any other changes that need to be made to make it appear correctly at different font sizes.

## 💰 Use value

This is important from an accessibility perspective as many users may want to set their browser font size to one that is different from the default to make information easier for them to read. The usability of a lot the guidance site will be affected if it doesn't accommodate a different font size by resizing accordingly.
